src/components/ha-navigation-picker.ts

```diff
@@ -0,0 +1,220 @@
+  icon: panel.icon ?? "mdi:view-dashboard",
+  title:
+    panel.url_path === hass.defaultPanel
+      ? hass.localize("panel.states")
+      : hass.localize(`panel.${panel.title}`) ||
+        panel.title ||
+        (panel.url_path ? titleCase(panel.url_path) : ""),
+});
+
+@customElement("ha-navigation-picker")
+export class HaNavigationPicker extends LitElement {
+  @property() public hass?: HomeAssistant;
+
+  @property() public label?: string;
+
+  @property() public value?: string;
+
+  @property() public helper?: string;
+
+  @property({ type: Boolean }) public disabled = false;
+
+  @property({ type: Boolean }) public required = false;
+
+  @state() private _opened = false;
+
+  private navigationItemsLoaded = false;
+
+  private navigationItems: NavigationItem[] = DEFAULT_ITEMS;
+
+  @query("ha-combo-box", true) private comboBox!: HaComboBox;
+
+  protected render(): TemplateResult {
+    return html`
+      <ha-combo-box
+        .hass=${this.hass}
+        item-value-path="path"
+        item-label-path="path"
+        .value=${this._value}
+        allow-custom-value
+        .filteredItems=${this.navigationItems}
+        .label=${this.label}
+        .helper=${this.helper}
+        .disabled=${this.disabled}
+        .required=${this.required}
+        .renderer=${rowRenderer}
+        @opened-changed=${this._openedChanged}
+        @value-changed=${this._valueChanged}
+        @filter-changed=${this._filterChanged}
+      >
+      </ha-combo-box>
+    `;
+  }
+
+  private async _openedChanged(ev: PolymerChangedEvent<boolean>) {
+    this._opened = ev.detail.value;
+    if (this._opened && !this.navigationItemsLoaded) {
+      this._loadNavigationItems();
+    }
+  }
+
+  private async _loadNavigationItems() {
+    const panels = Object.entries(this.hass!.panels).map(([id, panel]) => ({
+      id,
+      ...panel,
+    }));
+    const lovelacePanels = panels.filter(
+      (panel) => panel.component_name === "lovelace"
+    );
+
+    const viewConfigs = await Promise.all(
+      lovelacePanels.map((panel) =>
+        fetchConfig(
+          this.hass!.connection,
+          // path should be null to fetch default lovelace panel
+          panel.url_path === "lovelace" ? null : panel.url_path,
+          true
+        )
+          .then((config) => [panel.id, config] as [string, LovelaceConfig])
+          .catch((_) => [panel.id, undefined] as [string, undefined])
+      )
+    );
+
+    const panelViewConfig = new Map(viewConfigs);
+
+    this.navigationItems = [];
+
+    for (const panel of panels) {
+      this.navigationItems.push(createPanelNavigationItem(this.hass!, panel));
+
+      const config = panelViewConfig.get(panel.id);
+
+      if (!config) continue;
+
+      const viewItems = config.views.map<NavigationItem>((view, index) =>
+        createViewNavigationItem(panel.url_path, view, index)
+      );
+      this.navigationItems = this.navigationItems.concat(viewItems);
+    }
+
+    this.navigationItemsLoaded = true;
```

Comment:
We should set this to true as soon as possible, so we won't run this twice when quickly opening and closing